From 8956084727eefbb2bf7eddacd50dacd7930c8ddb Mon Sep 17 00:00:00 2001
From: Michael Weiser <michael.weiser@atos.net>
Date: Tue, 26 Oct 2021 11:11:48 +0200
Subject: [PATCH 4/9] cifs.upcall: Switch to RFC principal type naming

Switch from old-style MIT krb5 gss_nt_service_name principal type
constant name to the now preferred GSS_C_NT_HOSTBASED_SERVICE.

Signed-off-by: Michael Weiser <michael.weiser@atos.net>
---
 cifs.upcall.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cifs.upcall.c b/cifs.upcall.c
index 5e7c0a1..7341c2f 100644
--- a/cifs.upcall.c
+++ b/cifs.upcall.c
@@ -798,7 +798,7 @@ cifs_gss_get_req(const char *host, DATA_BLOB *mechtoken, DATA_BLOB *sess_key)
 	target_name_buf.length = service_name_len;
 
 	maj_stat = gss_import_name(&min_stat, &target_name_buf,
-			(gss_OID)gss_nt_service_name, &target_name);
+			GSS_C_NT_HOSTBASED_SERVICE, &target_name);
 	free(service_name);
 	if (GSS_ERROR(maj_stat)) {
 		cifs_gss_display_status("gss_import_name", maj_stat, min_stat);
-- 
2.35.1

